include/oneapi/dpl/pstl/utils.h

```diff
@@ -646,6 +646,61 @@ __dpl_ceiling_div(_T1 __number, _T2 __divisor)
     return (__number - 1) / __divisor + 1;
 }
 
+// TODO In C++20 we may try to use std::equality_comparable
+template <typename _Iterator1, typename _Iterator2, typename = void>
+struct __is_equality_comparable : std::false_type
+{
+};
+
+// All with implemented operator ==
+template <typename _Iterator1, typename _Iterator2>
+struct __is_equality_comparable<
+    _Iterator1, _Iterator2,
+    std::void_t<decltype(::std::declval<::std::decay_t<_Iterator1>>() == ::std::declval<::std::decay_t<_Iterator2>>())>>
+    : std::true_type
+{
+};
+
+#if _ONEDPL_BACKEND_SYCL
```

Comment:
Could there ever be a need to use `__iterators_possible_equal` with the sycl iterator specialization elsewhere in the codebase? If so, I think it would be best to keep its implementation in a utility header, so it could be used in other places without including `numeric_impl_hetero.h`. Is the concern with adding the SYCL specific code into `utils.h`?

Do any others have an opinion on this?

Reply:
The idea is that if we use sycl iterator, the header file `include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h` already included. For other types of iterator this code is not needed.

Reply:
Sounds good. I will make this change then.

Reply:
yes, SYCL specific code is allow from `.../hetero/dpcpp` level.